### ISSUE TITLE
Support different types for hash key and range key

### ISSUE THREAD

**Issue by cb372** [guardian/scanamo#4]
`get`, `delete` and so on take the key as `(key: (Symbol, K)*)`, but this means that all the components of the primary key have to have the same type `K`. This seems like quite a special case to me.

For example, in my app, the hash key is a `RecipeId` (with a DynamoFormat to persist it as a string) and the range key is the build number, an `Int`.

What about just providing 2 versions of each method that requires a key? e.g.

```scala
def get[K, T](client: AmazonDynamoDB)(tableName: String)(hashkey: (Symbol, K))
    (implicit fk: DynamoFormat[K], ft: DynamoFormat[T]): Option[ValidatedNel[DynamoReadError, T]] = ...

def get[HK, RK, T](client: AmazonDynamoDB)(tableName: String)(hashkey: (Symbol, HK), rangekey: (Symbol, RK))
    (implicit fhk: DynamoFormat[HK], frk: DynamoFormat[RK], ft: DynamoFormat[T]): Option[ValidatedNel[DynamoReadError, T]] = ...
```

The varargs param strikes me as a bit odd anyway, as you will only ever pass exactly 1 or 2 keys.

I can put a PR together if you're happy with this suggestion.

**Comment by adamnfish** [comment]
:+1:

**Comment by LATaylor-guardian** [comment]
:+1:

**Comment by philwills** [comment]
This is addressed along with one or two other things by #8

**Comment by philwills** [comment]
I've just made a 0.2.0 release which supports different types for hash and range keys on get/put/delete.
